docs/source/asyncio.rst

```diff
@@ -0,0 +1,134 @@
+Asyncio
+=======
+
+Support for asyncio is largely built on top of the `Alembic Cookbook`_ example,
+inlined here for posterity:
+
+.. code-block:: python
+
+   import asyncio
+
+   # ... no change required to the rest of the code
+   
+   def do_run_migrations(connection):
+       context.configure(connection=connection, target_metadata=target_metadata)
+   
+       with context.begin_transaction():
+           context.run_migrations()
+   
+   
+   async def run_migrations_online():
+       """Run migrations in 'online' mode.
+   
+       In this scenario we need to create an Engine
+       and associate a connection with the context.
+   
+       """
+       connectable = AsyncEngine(
+           engine_from_config(
+               config.get_section(config.config_ini_section),
+               prefix="sqlalchemy.",
+               poolclass=pool.NullPool,
+               future=True,
+           )
+       )
+   
+       async with connectable.connect() as connection:
+           await connection.run_sync(do_run_migrations)
+
+       await connectable.run_sync(do_run_migrations)
+   
+   
+   if context.is_offline_mode():
+       run_migrations_offline()
+   else:
+       asyncio.run(run_migrations_online())
+
+Note that this is a prerequisite for how one gets **alembic itself** to run with an async
+connection, when running ``alembic`` commands interatively yourself.
+
+At this point, you just need to make sure the ``alembic_engine`` fixture is producing a async engine.
+something like
+
+.. code-block:: python
+
+   from sqlalchemy import create_engine
+   from sqlalchemy.ext.asyncio import create_engine_async, AsyncEngine
```

Comment:
Do we want to include this `AsyncEngine` import in the above example as well?  At least for completeness.

Reply:
I copied that example directly from alembic's docs, so perhaps not, so i dont have to maintain a diff, if it ever changes (and it's going [to change](https://github.com/sqlalchemy/alembic/pull/973))

Reply:
we can update it in the alembic docs too.

Reply:
added here https://github.com/sqlalchemy/alembic/commit/1364bbf9b782178cad66ca22783dde1ba8353fde